crates/runtime/src/component/mod.rs

```diff
@@ -271,6 +365,70 @@ pub type InvocationStream = Pin<
     >,
 >;
 
+impl<H> Component<H>
+where
+    H: MinimalHandler,
+{
+    /// Compiles a WebAssembly component using [Runtime]. Modules are not supported with this function
+    ///
+    /// If you need to pass a custom context, use [CustomCtxComponent::new_with_linker_minimal].
+    ///
+    /// It is expected that the `linker_fn` is used to link any and all imports to the component. If
+    /// not, then the instantiation of the instance will fail.
+    #[instrument(level = "trace", skip_all)]
+    pub fn new_with_linker_minimal(
```

Comment:
Nit: This is almost the same as the other new function. Maybe pull out the reusable bits if possible?